### DISCUSSION TITLE
Error 301

### DISCUSSION THREAD

**Discussion by Charles9424** [AndresRamos/ARSoftware.Cfdi.DescargaMasiva#7] [Q&A]
Marca el error 301 incluso con el certificado de prueba, que el XML se esta mandando erroneo

**Comment by AndresRamos** [comment]
Todavía te sigue marcando este error?

**Reply by Charles9424** [reply]
Hola buenas tardes Andres, en efecto pero ahora solo al momento de consultar por uuid ya probe varias formas y no he podido llegar a un  solucion

**Reply by AndresRamos** [reply]
Si funcionan las consultas por fecha, solamente te marca error al buscar por UUID?

**Reply by Charles9424** [reply]
Si despues de un rato y varias prueba puede hacer consulta con las fechas, al momento de quitar todo  y madar solamente la uuid me marca varios errors 

<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/"><s:Body><s:Fault><faultcode xmlns:a="http://schemas.microsoft.com/net/2005/12/windowscommunicationfoundation/dispatcher">a:InternalServiceFault</faultcode><faultstring xml:lang="en-US">Error in deserializing body of request message for operation 'SolicitaDescarga'.</faultstring><detail><ExceptionDetail xmlns="http://schemas.datacontract.org/2004/07/System.ServiceModel" xmlns:i="http://www.w3.org/2001/XMLSchema-instance"><HelpLink i:nil="true"/><InnerException><HelpLink i:nil="true"/><InnerException><HelpLink i:nil="true"/><InnerException i:nil="true"/><Message>The string '' is not a valid AllXsd value.</Message><StackTrace>   at System.Xml.Schema.XsdDateTime..ctor(String text, XsdDateTimeFlags kinds)&#xD;
   at System.Xml.XmlConvert.ToDateTime(String s, XmlDateTimeSerializationMode dateTimeOption)&#xD;
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderISolicitaDescargaService.Read18_SolicitudDescargaMasivaTercero(Boolean isNullable, Boolean checkType)&#xD;
   at Microsoft.Xml.Serialization.GeneratedAssembly.XmlSerializationReaderISolicitaDescargaService.Read21_SolicitaDescarga()&#xD;
   at System.Xml.Serialization.XmlSerializer.Deserialize(XmlReader xmlReader, String encodingStyle, XmlDeserializationEvents events)</StackTrace><Type>System.FormatException</Type></InnerException><Message>There is an error in XML document (1, 366).</Message><StackTrace>   at System.Xml.Serialization.XmlSerializer.Deserialize(XmlReader xmlReader, String encodingStyle, XmlDeserializationEvents events)&#xD;
   at System.ServiceModel.Dispatcher.XmlSerializerOperationFormatter.DeserializeBody(XmlDictionaryReader reader, MessageVersion version, XmlSerializer serializer, MessagePartDescription returnPart, MessagePartDescriptionCollection bodyParts, Object[] parameters, Boolean isRequest)</StackTrace><Type>System.InvalidOperationException</Type></InnerException><Message>Error in deserializing body of request message for operation 'SolicitaDescarga'.</Message><StackTrace>   at System.ServiceModel.Dispatcher.XmlSerializerOperationFormatter.DeserializeBody(XmlDictionaryReader reader, MessageVersion version, XmlSerializer serializer, MessagePartDescription returnPart, MessagePartDescriptionCollection bodyParts, Object[] parameters, Boolean isRequest)&#xD;
   at System.ServiceModel.Dispatcher.XmlSerializerOperationFormatter.DeserializeBody(XmlDictionaryReader reader, MessageVersion version, String action, MessageDescription messageDescription, Object[] parameters, Boolean isRequest)&#xD;
   at System.ServiceModel.Dispatcher.OperationFormatter.DeserializeBodyContents(Message message, Object[] parameters, Boolean isRequest)&#xD;
   at System.ServiceModel.Dispatcher.OperationFormatter.DeserializeRequest(Message message, Object[] parameters)&#xD;
   at System.ServiceModel.Dispatcher.DispatchOperationRuntime.DeserializeInputs(MessageRpc&amp; rpc)&#xD;
   at System.ServiceModel.Dispatcher.DispatchOperationRuntime.InvokeBegin(MessageRpc&amp; rpc)&#xD;
   at System.ServiceModel.Dispatcher.ImmutableDispatchRuntime.ProcessMessage5(MessageRpc&amp; rpc)&#xD;
   at System.ServiceModel.Dispatcher.ImmutableDispatchRuntime.ProcessMessage11(MessageRpc&amp; rpc)&#xD;
   at System.ServiceModel.Dispatcher.MessageRpc.Process(Boolean isOperationContextSet)</StackTrace><Type>System.ServiceModel.CommunicationException</Type></ExceptionDetail></detail></s:Fault></s:Body></s:Envelope>

**Reply by Charles9424** [reply]
Tienes una idea del porque me sale este error o como es que tengo que mardar lo solicitud de busqueda por uuid porque segun la documentacion no debo declara ni el rango de fechas ni el emisor ni solicitante

**Reply by AndresRamos** [reply]
Acabo de crear un nuevo issue #9
Vamos a darle seguimiento ahi, necesito que compartas el codigo que estas uando para enviar la solicitud y los XML que genera.

**Comment by juliancabcanul** [comment]
Saludos Andres, en efecto ese error se sigue generando. De que forma se podria debuggear para revisar que esta mal? Agradeceremos tu apoyo.

**Comment by juliancabcanul** [comment]
![Imagen de WhatsApp 2023-03-14 a las 15 56 28](https://user-images.githubusercontent.com/87134597/225153382-d89aa668-7b7a-4a11-a200-b458dce18e20.jpg)

**Reply by AndresRamos** [reply]
Podrias compartir el XML que se genera?

**Reply by Charles9424** [reply]
En las pruebas que hice siempre tuve que mandar dos campos llenos  de los rfc ya sea el emisor, receptor, y solicitante

**Reply by AndresRamos** [reply]
Este es de la aplicación [Manejador Documentos CFDI](https://github.com/AndresRamos/ARSoftware.ManejadorDocumentosCfdi) que utiliza esta utilería para hacer las descargos.

Creo que aquí el problema es que no esta asignando el RFC Solicitante. Este campo se asigna en la configuración de la empresa.

**Comment by juliancabcanul** [comment]
Esta es la solicitud:

```xml
<S11:Envelope xmlns:S11="http://schemas.xmlsoap.org/soap/envelope/" xmlns:des="http://DescargaMasivaTerceros.sat.gob.mx" xmlns:ds="http://www.w3.org/2000/09/xmldsig#">
	<S11:Header />
	<S11:Body>
		<des:SolicitaDescarga>
			<des:solicitud FechaInicial="2023-03-14T00:00:00" FechaFinal="2023-03-14T23:59:59" RfcEmisor="" RfcSolicitante="" TipoSolicitud="CFDI" RfcACuentaTerceros="">
				<des:RfcReceptores>
					<des:RfcReceptor>CMM920924K74</des:RfcReceptor>
				</des:RfcReceptores>
				<Signature xmlns="http://www.w3.org/2000/09/xmldsig#">
					<SignedInfo>
						<CanonicalizationMethod Algorithm="http://www.w3.org/TR/2001/REC-xml-c14n-20010315" />
						<SignatureMethod Algorithm="http://www.w3.org/2000/09/xmldsig#rsa-sha1" />
						<Reference URI="">
							<Transforms>
								<Transform Algorithm="http://www.w3.org/2000/09/xmldsig#enveloped-signature" />
							</Transforms>
							<DigestMethod Algorithm="http://www.w3.org/2000/09/xmldsig#sha1" />
							<DigestValue>t5ZZFBFpWRD+RuRReLZumx8hfi4=</DigestValue>
						</Reference>
					</SignedInfo>
					<SignatureValue>go4AUtUOINxfByLi46+0CSK0PPLONo0+vuE9aQ+1RXCtbw78ltFUPmRkcwGw3+mJTDsRnvNYdQuBMSg6v+hQA9tVg9lDKa0IU5eRc5vWWIUMYa883y//6Fqsk55hjtGhgiGiOt/rAJV8EVVSU59xFsYk4kJJzQmpWF0FwjM6uV6TFL+afHS8tvB8IHftv91J4TLCUiW+4AU95IPBoTr3jm3vAyinnfLlK5UR0Q48ck171RfFLWZ7hI10WRZQnvBYZat9va6nsGmxoelfHUtlIHPFP4tiTrWqO5o0F7lesxq7xM7SVRILqGC1kbMhkHgjD36vae9BTqeRKFjBXUWddA==</SignatureValue>
					<KeyInfo>
						<X509Data>
							<X509IssuerSerial>
								<X509IssuerName>OID.1.2.840.113549.1.9.2=responsable: ADMINISTRACION CENTRAL DE SERVICIOS TRIBUTARIOS AL CONTRIBUYENTE, OID.2.5.4.45=SAT970701NN3, L=CUAUHTEMOC, S=CIUDAD DE MEXICO, C=MX, PostalCode=06300, STREET="AV. HIDALGO 77, COL. GUERRERO", E=contacto.tecnico@sat.gob.mx, OU=SAT-IES Authority, O=SERVICIO DE ADMINISTRACION TRIBUTARIA, CN=AUTORIDAD CERTIFICADORA</X509IssuerName>
								<X509SerialNumber>275106190557734483187066766829380168430195585078</X509SerialNumber>
							</X509IssuerSerial>
							<X509Certificate>MIIGPzCCBCegAwIBAgIUMDAwMDEwMDAwMDA1MDQ1NTg0MDYwDQYJKoZIhvcNAQELBQAwggGEMSAwHgYDVQQDDBdBVVRPUklEQUQgQ0VSVElGSUNBRE9SQTEuMCwGA1UECgwlU0VSVklDSU8gREUgQURNSU5JU1RSQUNJT04gVFJJQlVUQVJJQTEaMBgGA1UECwwRU0FULUlFUyBBdXRob3JpdHkxKjAoBgkqhkiG9w0BCQEWG2NvbnRhY3RvLnRlY25pY29Ac2F0LmdvYi5teDEmMCQGA1UECQwdQVYuIEhJREFMR08gNzcsIENPTC4gR1VFUlJFUk8xDjAMBgNVBBEMBTA2MzAwMQswCQYDVQQGEwJNWDEZMBcGA1UECAwQQ0lVREFEIERFIE1FWElDTzETMBEGA1UEBwwKQ1VBVUhURU1PQzEVMBMGA1UELRMMU0FUOTcwNzAxTk4zMVwwWgYJKoZIhvcNAQkCE01yZXNwb25zYWJsZTogQURNSU5JU1RSQUNJT04gQ0VOVFJBTCBERSBTRVJWSUNJT1MgVFJJQlVUQVJJT1MgQUwgQ09OVFJJQlVZRU5URTAeFw0yMDA3MjIxOTE5NThaFw0yNDA3MjIxOTIwMzhaMIHbMR4wHAYDVQQDExVNRVpGRVIgVFJBREUgU0EgREUgQ1YxHjAcBgNVBCkTFU1FWkZFUiBUUkFERSBTQSBERSBDVjEeMBwGA1UEChMVTUVaRkVSIFRSQURFIFNBIERFIENWMQswCQYDVQQGEwJNWDElMCMGCSqGSIb3DQEJARYWbGF1cmFnbXJfMUBob3RtYWlsLmNvbTElMCMGA1UELRMcTVRSMDUwMzAxTUIyIC8gUkVHRzQ4MDUwOThBMDEeMBwGA1UEBRMVIC8gUkVHRzQ4MDUwOUhERkJOTDA5MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAnWtj7sxFzlblzoJ5iIkvilkyQDiLA4yUSw0LGr7a9bbomEcuT3keIPuXOdQYBW8/ahQ0bQ3/BpLE3GjxLOhLX+ZmKB/QiH5DToRkocXgH5S6t1jJblayPoXTQHsIJRCOI1wlG1NDv5oCTAg3svld8BhAr3i1EDDCQMclprjj68NqbCRYqm9/DMbDXJmzvDbQeLHu3TmT22dAewPQYFahHnip1pBYPSK27n1PBJYaCnntqpZ73sVIHWlR8wpKqQCtnAxQ+zZC1/rgrWNJXaPA3YbQ+x8ujL0wVSzwWtVbN4yxHD1YugPgWyyDjGl/zmkAzD0RurLeBis6TiYAsP+AxQIDAQABo08wTTAMBgNVHRMBAf8EAjAAMAsGA1UdDwQEAwID2DARBglghkgBhvhCAQEEBAMCBaAwHQYDVR0lBBYwFAYIKwYBBQUHAwQGCCsGAQUFBwMCMA0GCSqGSIb3DQEBCwUAA4ICAQBFHb0zx/f0KLC0F/7gOu+ssg6L4TKBhFQfkWqm/5VmLPG3R9dOTKDAwJYRzE8W5Dym6SSJ1QxF3MpBZQnKAfTtoCWFRVSxIKwBCW2lHLd67g4ZV49VamRdNUVEKwNg1QI+6+gUZWH0/nxd6MIWojRx2EpBLY0e1yew1pJsmV5EMnqPOZCVOv4ZrQKjnTTyPovDRKaQgFjqS8eYypeX3papyi+AG70hsj+Ij3qKnoUTcNUIM/AzJhzI9nUV9Oxf4Tpj7/UQaEGmgakvUKtp+lTT9sETuDrK1NQmzVVqPGMErQFLFdPhrLryRJu1iU7gtvTeNEjPmXPAjPNQg5aQGHWYITId7UFqlYFqf4t1YEDlaqpEtaJdKofDA3zR/lQicpQoMEF75MNPHzK8ySKd3cImnUppWbw1TgmcjoofLHPmjvJdH5GS6Qt63+vhGR4dpdYHpZFTTgblBp3PDiUUotRiVr0dwnmIImvXUGmVS93pKU3KpMVJBz1+50GtTMR58RS1HB1QbCEOax93d9aTGb5iGBklQbLH0P25U6KJfSCLdnXI3/BTqOFlH2STzcVuNCZL1dwr9bLu7eP7MEubjEo1m1qAMfPShxxAODVIPs5Djy0U0Zt4zWGgqyHeyvfYrNTws1QOyZgNs1X6q68nge7jRU56oB895OQEZA1HfBdScQ==</X509Certificate>
						</X509Data>
					</KeyInfo>
				</Signature>
			</des:solicitud>
		</des:SolicitaDescarga>
	</S11:Body>
</S11:Envelope>
```

Esta es la respuesta:

```xml
<s:Envelope xmlns:s="http://schemas.xmlsoap.org/soap/envelope/">
	<s:Body xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns:xsd="http://www.w3.org/2001/XMLSchema">
		<SolicitaDescargaResponse xmlns="http://DescargaMasivaTerceros.sat.gob.mx">
			<SolicitaDescargaResult CodEstatus="301" Mensaje="XML Mal Formado" />
		</SolicitaDescargaResponse>
	</s:Body>
</s:Envelope>
```

**Reply by AndresRamos** [reply]
Puedo ver que no viene el RfcSolicitante. Este campo es obligatorio.

**Comment by juliancabcanul** [comment]
Yo asi las genero y no tengo problemas. Recién publico una actualización Andres y esta funcional.

 

Saludos

 



 

De: Charles9424 ***@***.***> 
Enviado el: jueves, 16 de marzo de 2023 10:40 a. m.
Para: AndresRamos/ARSoftware.Cfdi.DescargaMasiva ***@***.***>
CC: Julian Cab Canul ***@***.***>; Comment ***@***.***>
Asunto: Re: [AndresRamos/ARSoftware.Cfdi.DescargaMasiva] Error 301 (Discussion #7)

 

Tienes una idea del porque me sale este error o como es que tengo que mardar lo solicitud de busqueda por uuid porque segun la documentacion no debo declara ni el rango de fechas ni el emisor ni solicitante
